### ISSUE TITLE
Provide guidance on setting up an IDistributedCache at the host level for BackgroundServices

### ISSUE THREAD

**Issue by ShaneCourtrille** [OrchardCMS/OrchardCore#14625]
Thanks to the Redis module we already have IDistributedCache available in the tenant DI Containers but I'm wondering how we would safely set up an IDistributedCache in the root level DI container due to the fact the root gets cloned for the tenant (right? like 90% sure on this).  

Is unregistering IDistributedCache before the OrchardCore Redis Startup runs good enough?

Really trying to avoid cross-tenant data spillage since that would obviously be... bad.

**Comment by MikeAlhayek** [comment]
The host tenant is just like any other tenant. So if you resource the `IDistributedCache` from it's tenant, then you should get an instance and they keys usually are prefixed with the correct tenant name.

I usually set `InstancePrefix` option when configuring Redis to allow me to use the same Redis instance for multiple environments.

**Comment by jtkech** [comment]
> Is unregistering IDistributedCache before the OrchardCore Redis Startup runs good enough?

At the tenant level, even if Redis is not enabled we re-register `IDistributedCache`, so it is always a different implementation instance per tenant and different from the host one.

So you can do what you want at the host level.

For info we have a `DistributedShellHostedService` which is a hosted `BackgroundService` used to keep in sync tenant states, it uses the possible `IDistributedCache` of the Default tenant, we did this to not have to register a distributed cache implementation at the host level, but we have to clone the Default tenant in a special way to not interfere with our `ShellHost`, so this is not recommended.

**Comment by ShaneCourtrille** [comment]
@jtkech Yeah I've taken a look at that thing and it's complicated so definitely wanted to avoid that.  Thanks for the info.

**Comment by ShaneCourtrille** [comment]
@jtkech Would there be any interest in a new DistributedBackgroundService base class that contains all of the code necessary to use the Default tenant's IDistributedCache? I created one for our solution based on the existing distributed tenant sync.
